### PR TITLE
Feature/auth refactor

### PR DESCRIPTION
**Pull request by joe94**
Changes to be made:

- [x] Change instances of `JWT` to Bearer

- [x] Create a new type of fixed token type of Bearer Acquirer

- [x] Provide implementations through constructors by only exporting necessary structs and functions 

**Further thoughts:** I think our constructors (i.e. `NewBasicAuthAcquirer(..)`) should also return an error value.  
**Benefits:**
- fail fast 
- remove some edge cases in the `Acquire()` method

- [x] Add a few more unit test cases

### CODE REVIEW

**Comment by joe94** [comment]
This fixes #29

**Comment by joe94** [comment]
Based on reviews, let's do the following:
- Unify constant token `Basic` and `Bearer` under a `FixedToken` constructor 
- Continue the change laid out in the todo list above

**Comment by joe94** [comment]
Notable changes:

-Removed the basic.go file as the functionality is covered by the fixedTokenAcquirer
The removed plaintext constructor is an easy one-liner on library user code

-Made the `time.Unix(0,0)` == expiringDate special case more explicit to library users

File: acquire/auth.go [hunk 1]
```diff
@@ -8,31 +8,64 @@ import (
...
+}
+
+func (f *fixedValueAcquirer) Acquire() (string, error) {
+	return f.AuthValue, nil
+}
+
+//NewFixedAuthAcquirer returns an acquirer with a fixed authentication
+//value. 'authValue' should be the full authorization value of the form '[type] [token]'
+//(i.e. Bearer xyz)
+func NewFixedAuthAcquirer(authValue string) (Acquirer, error) {
```
Comment (kristinaspring) [inline acquire/auth.go]: I think we should return a concrete type...
https://github.com/golang/go/wiki/CodeReviewComments#interfaces

the `AuthValue` of the fixedValueAcquirer should be private, but the struct itself can be public.

File: acquire/auth_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"net/http/httptest"
+	"testing"
+
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAddAuth(t *testing.T) {
+
+	t.Run("RequestIsNil", func(t *testing.T) {
```
Comment (kristinaspring) [inline acquire/auth_test.go]: Can you turn these tests into table tests?

File: acquire/bearer.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	"time"
+
+	"github.com/goph/emperror"
+)
+
+//TokenParser defines the function signature of a bearer token extractor from a payload
+type TokenParser func([]byte) (string, error)
+
+//ParseExpiration defines the function signature of a bearer token expiration date extractor
+type ParseExpiration func([]byte) (time.Time, error)
```
Comment (kristinaspring) [inline acquire/bearer.go]: Not sure why this type line and some other function signatures in this file aren't getting colored correctly? I can't find any typos/potential parsing issues.

File: acquire/bearer.go [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+}
+
+//SimpleBearer defines the field name mappings used by the default bearer token and expiration parsers
+type SimpleBearer struct {
+	ExpiresInSeconds float64 `json:"expires_in"`
+	Token            string  `json:"serviceAccessToken"`
+}
+
+//NewRemoteBearerTokenAcquirer returns an acquirerr which fetches tokens from a configurable URL location
+//The acquirerr caches tokens and only re-fetches them from such URL once they have expired
```
Comment (kristinaspring) [inline acquire/bearer.go]: These comments become GoDocs, where paragraphs are only made if there are two enters.  These comments will read like this:
> NewRemoteBearerTokenAcquirer returns an acquirerr which fetches tokens from a configurable URL locationThe acquirerr caches tokens and only re-fetches them from such URL once they have expiredote: If you'd like for a token to never expire, set is expiration to time.Unix(0,0)

Comments should read like full sentences, because they will be put into the generated GoDoc, and if you want something to be separated into another paragraph/line, it should have an empty comment line between the two paragraphs.

Comment (kristinaspring) [reply acquire/bearer.go]: In a similar vein, I find it more difficult to read comments when the words start right next to the `//`, which is why I prefer if there is a space between them, but that's more of a personal preference.

File: acquire/bearer.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+//Note: If you'd like for a token to never expire, set is expiration to time.Unix(0,0)
```
Comment (kristinaspring) [inline acquire/bearer.go]: It's not advisable to have a token that doesn't expire.  I would suggest either saying so in the comment or removing this tip.

File: acquire/bearer.go [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+func NewRemoteBearerTokenAcquirer(options RemoteBearerTokenAcquirerOptions) (Acquirer, error) {
+	if options.GetToken == nil {
+		options.GetToken = DefaultTokenParser
+	}
+
+	if options.GetExpiration == nil {
+		options.GetExpiration = DefaultExpirationParser
+	}
+
+	//TODO: we should inject defaults values for the other options as well
```
Comment (kristinaspring) [inline acquire/bearer.go]:
```suggestion
	//TODO: we should inject timeout and buffer defaults values as well
```

File: acquire/bearer.go [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+		return "", emperror.Wrap(err, "failed to create new request for Bearer")
+	}
+
+	for key, value := range acquirer.options.RequestHeaders {
+		req.Header.Set(key, value)
+	}
+
+	resp, errHTTP := acquirer.httpClient.Do(req)
+	if errHTTP != nil {
+		return "", fmt.Errorf("error acquiring bearer token: [%s]", errHTTP.Error())
```
Comment (kristinaspring) [inline acquire/bearer.go]: Can we wrap this error and below errors with emperror instead of using fmt.Errorf?  Can also add some context, like the URL we were trying to hit.
